schemachange/tenant: destroy tenant data with GC job

spaskob opened the pull request:
Fixes #48775.

This PR adds tenant GC fields to schema change Details and
Progress protos and adds support for deleting tenant's data
via the schema change GC job.

Release note: none.

Commit:
sql: changes to tenant API

Informs #48775.

This PR:
 - adds `DeleteTenantRecord` to remove a tenant record from the tenants
table.
 - makes `GetTenantRecord` public.

Release note: none.

Commit:
schemachange/tenants: add tenant to GC job

Informs #48775.

This PR adds tenant GC fields to schema change Details and
Progress protos and adds support for deleting tenant's data
via the schema change GC job.

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55756)
<!-- Reviewable:end -->

Commit:
sql/tenant: use async GC job for tenant

This commit creates a GC tenant job in the end of the destroy tenant
operation.

Release note: none.

pbardea reviewed (approved):
> Reviewed 1 of 1 files at r1, 1 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55756)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @lucy-zhang, and @pbardea)



<!-- Sent from Reviewable.io -->

thoszhang reviewed (approved):
> Reviewed 1 of 1 files at r1, 7 of 9 files at r4, 2 of 4 files at r5, 5 of 5 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55756)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt)



<!-- Sent from Reviewable.io -->

spaskob commented:
bors r+
